src/library/trackcollection.cpp

```diff
@@ -510,4 +514,80 @@ StorageUnicodeFunctions::likeFunction(sqlite3_context *p,
                                     itrString, endString, E));
 }
 */
+//Private function
+
+bool TrackCollection::settingsPathStatus(){
+    QFileInfo fle(m_pConfig->getSettingsPath());
+    QFile::Permissions permit = fle.permissions();
+    permit = permit & MASK;
+    const short int NONE = QFile::NoOptions;
+    const short int EXE = QFile::ExeOwner;
+    const short int WRITE = QFile::WriteOwner;
+    const short int WRITE_EXE = QFile::WriteOwner | QFile::ExeOwner;
+    const short int READ = QFile::ReadOwner;
+    const short int READ_EXE = QFile::ReadOwner | QFile::ExeOwner;
+    const short int READ_WRITE = QFile::ReadOwner | QFile::WriteOwner;
+    switch(permit)
```

Comment:
But as I try to think of it, the case like if(!(permit & QFile::ReadUser)) and the case of if(!(permit & QFile::ReadUse & QFile::WriteUser)) will make the same, if the 2nd case is true then automatically the first case will be true, may be this mechanism is good, in this sencse other wise the code in the if will become long, if I need to make it happen to all the cases